src/features/session_trace/aggregate/trace/storage.js

```diff
@@ -139,6 +141,16 @@ export class TraceStorage {
     this.storeTiming({ [name]: value })
   }
 
+  isBeforeSessionExpiry (entryTimestamp) {
+    let isValidTimingEntry = entryTimestamp < DEFAULT_EXPIRES_MS
+
+    if (!isValidTimingEntry && !this.eventsSeenAfterSessionExpire) {
+      this.eventsSeenAfterSessionExpire = true
+    }
+
+    return isValidTimingEntry
```

Comment:
Something like this might be less verbose
```suggestion
    return this.eventsSeenAfterSessionExpire ||= entryTimestamp < DEFAULT_EXPIRES_MS
```